### DISCUSSION TITLE
Issue with Webservice and loading of .js files

### DISCUSSION THREAD

**Discussion by mjapple** [YetiForceCompany/YetiForceCRM#17069] [General] [answered]
Hello Everyone and Yetiforce Team!
Thanks foe all you efforts and I would like to report an issue or could be some point which I don't know yet how to solve it.
While trying to setting up the "WebserviceStandard" I got to a point where the CRM will load not complete! In. the browser console I observe many errors. It seems no .js file could be loaded.
Here are my settings and the environment on my end.
![TFI Tools_Proprety 2023-05-08 at 8 21 37 PM](http://user-images.githubusercontent.com/6837503/236892290-e8064533-a01d-423c-9d91-e5c5fe284f36.png)

![TFI Tools_Proprety 2023-05-08 at 8 21 50 PM](https://user-images.githubusercontent.com/6837503/236892255-094c9c9f-62e8-4d67-bec3-e24d39af9b2a.png)

![TFI Tools_Proprety 2023-05-08 at 8 21 58 PM](https://user-images.githubusercontent.com/6837503/236892243-034ffbfb-f9d4-422b-89f8-73aecbe282e8.png)
![TFI Tools_Proprety 2023-05-08 at 8 22 07 PM](https://user-images.githubusercontent.com/6837503/236892315-a8eeb7d3-a35f-4d9e-869c-fefed43f7b4f.png)
![TFI Tools_Proprety 2023-05-08 at 8 22 18 PM](https://user-images.githubusercontent.com/6837503/236892335-9485dd4d-85bb-41e0-b826-4b1cda2ae0b9.png)
![TFI Tools_Proprety 2023-05-08 at 8 22 44 PM](https://user-images.githubusercontent.com/6837503/236892347-cbee4021-b2e1-4dd7-acec-a07bbb1bb94d.png)
![TFI Tools_Proprety 2023-05-08 at 8 22 53 PM](https://user-images.githubusercontent.com/6837503/236892360-72022f0f-6586-4e32-97d0-a5e7bf19d88b.png)
![TFI Tools_Proprety 2023-05-08 at 8 23 02 PM](https://user-images.githubusercontent.com/6837503/236892373-68fa25ef-d46b-44b9-90da-31ef70937499.png)
![TFI Tools_Proprety 2023-05-08 at 8 23 12 PM](https://user-images.githubusercontent.com/6837503/236892390-53917902-fd10-4461-a696-ef930d0f7f27.png)
![TFI Tools_Proprety 2023-05-08 at 8 23 26 PM](https://user-images.githubusercontent.com/6837503/236892403-7ea99d05-0e16-442e-bc95-2c44995f6f6a.png)
![TFI Tools_Proprety 2023-05-08 at 8 23 49 PM](https://user-images.githubusercontent.com/6837503/236892413-70b4a84d-568f-45c1-96fa-f23e8e07a78c.png)
---
I started with the instructions in the link below: 
https://github.com/YetiForceCompany/YetiForcePortal2#-installation
I ran:
`yarn install --modules-folder "./public_html/libraries" --ignore-optional --production=true`
I ran this on my ___CRM___DIR:
`composer install --no-dev --no-interactionAfter these running composer on `
and then I figured out While I visit the crm no FontAwesome loads nor expandable menus.
I checked the browser Console and this is how it is 
![TFI Tools_Proprety 2023-05-08 at 8 24 46 PM](https://user-images.githubusercontent.com/6837503/236893422-dc4c457d-e766-4999-91ca-517d313a8e1c.png)
after success login this screen comes up:
![TFI Tools_Proprety 2023-05-08 at 8 25 06 PM](https://user-images.githubusercontent.com/6837503/236893509-22ac2e93-befd-4846-a9c6-afdedb745720.png)
and after login Error below shows up in system.log located in cache folder:

```
====================================================================================================================================
2023-05-08 20:25:03.9708 [error][[IllegalValue]] - purifyByType: 
2023-05-08 20:25:03.971 [error][] - ERR_NOT_ALLOWED_VALUE||
App\Exceptions\IllegalValue: ERR_NOT_ALLOWED_VALUE|| in /__CRM__DIR__/app/Purifier.php:557
Stack trace:
#0 /__CRM__DIR__/app/Request.php(170): App\Purifier::purifyByType()
#1 /__CRM__DIR__/modules/Users/actions/Login.php(178): App\Request->getByType()
#2 /__CRM__DIR__/modules/Users/actions/Login.php(131): Users_Login_Action->afterLogin()
#3 /__CRM__DIR__/modules/Users/actions/Login.php(72): Users_Login_Action->login()
#4 /__CRM__DIR__/include/main/WebUI.php(188): Users_Login_Action->process()
#5 /__CRM__DIR__/index.php(26): Vtiger_WebUI->process()
#6 /__CRM__DIR__/public_html/index.php(13): require('...')
#7 {main}
2023-05-08 20:25:03.8613 [info][[application]] - 
$_GET = ['module'=>'Users','action'=>'Login']
$_POST = ['_csrf'=>'sid:703df090ee54eef0898373f0b950ab484ae8e59,1683566672;ip:e0576f29f9d87b61e67bd22db3439528056ce86,1683566672','username'=>'USERNAMEXXXXX','password'=>'****','fingerprint'=>'']
$_COOKIE = ['YTSID'=>'sqj4jl2kt60h843s67miulosj']
$_SESSION = ['CSP_TOKEN'=>'bbb16eed4214708ec1564df838e33a9182abbcb09c0addd939db9812e8b566b','last_activity'=>1683566703.862001,'UserAuthMethod'=>'PASSWORD','authenticated_user_id'=>1,'app_unique_key'=>'d6f2f20b98cc91097b555b7c36ca8643a5c9050b','user_name'=>'USERNAMEXXXXX','full_user_name'=>'Dastouri MJ']
_HEADERS = ['Cookie'=>'YTSID=sqj4jl2kt60eh843s67miulosj','Accept-Language'=>'en-US,en;q=0.9','Accept-Encoding'=>'gzip, deflate, br','Sec-Fetch-Dest'=>'document','Sec-Fetch-User'=>'?1','Sec-Fetch-Mode'=>'navigate','Sec-Fetch-Site'=>'same-origin','Accept'=>'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7','User-Agent'=>'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36','Content-Type'=>'application/x-www-form-urlencoded','Origin'=>'null','Upgrade-Insecure-Requests'=>'1','Sec-Ch-Ua-Platform'=>'"macOS"','Sec-Ch-Ua-Mobile'=>'?0','Sec-Ch-Ua'=>'"Google Chrome";v="111", "Not(A:Brand";v="8", "Chromium";v="111"','Cache-Control'=>'max-age=0','Content-Length'=>'180','Connection'=>'keep-alive','Host'=>'systems.tfico.com','Authorization'=>'']
$libraries = ['ldap'=>'LBL_NO']
$database = ['character_set_server'=>'latin1']
$environment = ['currentUser'=>' (uid:1000)','openssl.cafile'=>false,'openssl.capath'=>false]
$headers = ['Header: server'=>'Apache']
====================================================================================================================================
```
and When I press Dashboard the browser shows the following errors. all are about .js files!

![TFI Tools_Proprety 2023-05-08 at 8 25 49 PM](https://user-images.githubusercontent.com/6837503/236893871-bf00c6ff-5b91-4cae-8de3-d9676c2ba8b2.png)

I tried to restart server and reload php and restart apache, emptying cache and cookies...

Would be nice if there is a solution for it
Many Thanks

**Comment by rskrzypczak** [comment, marked as answer]
I'm a bit confused because you're mixing YetiFroceCRM with YetiForcePortal2.
These are two different applications that should be located on separate servers.
It looks like you've run yarn and composer on a working YetiForceCRM version and messed up the libraries and hence these errors ;/
FYI: https://doc.yetiforce.com/introduction/installation-manual
      https://github.com/YetiForceCompany/YetiForcePortal2#-installation

The simplest solution to the problem will probably be to download the [complete version of YetiForceCRM](https://github.com/YetiForceCompany/YetiForceCRM/releases/tag/6.4.0) and restore the original directory ./public_html/libraries (and probably ./vendor too)

**Reply by mjapple** [reply]
Correct! In ./public_html/ "vendor" folder and "libraries" folder is replaced with original versions downloaded from the [Yetiforce Complete version](https://github.com/YetiForceCompany/YetiForceCRM/releases/tag/6.4.0). And all are fine!

**Reply by mjapple** [reply]
I replaced All(except Storage, user_privilages, config, Languages) files and folders with another production version 6.4.0 of Yetiforce and all are fine.
